Fix Incorrect Toggle Cvar (ammo)

FDAapproved opened the pull request:
Fixes a bug in the MENUDEFs for Hideous Helmet Extended, wherein the Ammo Counter menu option for font size was tied to whether or not the alcohol counter was enabled instead.

FDAapproved committed:
Fix Incorrect Toggle Cvar (ammo)

Ammo Counter Font Scale was tied to the alcohol counter font scale, which this commit fixes.

UndeadZeratul commented:
Oh, thank you!  I actually already fixed this in my development branch though, I should probably make a new release either way...

https://github.com/UndeadZeratul/hideous-helmet-extended/commit/5d250c88883a4bc4f9b7c21199d5ccccc3e081f2

FDAapproved committed:
Refactor: "I didn't push for a week" Edition

This started as an experiment to see how hard it would be to add units of measurement to the compass, but now has ballooned into a thought experiment for how to best implement dataclasses in zscript, if their primary purpose is information in -> information out + unique arithmetic. The compass itself has varying features planned, which I've reimplemented twice now, and am in the process of redoing again. Those features are as follows:
- Ability to convert MU coordinates into Meters, Kilometers, and Miles,
- Ability to display the cardinal directions the player is in relative to the origin coordinates of the map (x->East/West, y->North/South, z->Up/Down),
- Ability to display the X, Y, and Z labels for a position,
- Linebreaks in the positional string,
- A units label at the end of the position string.
This current revision comes as I realized adding padding via spaces both is jittery on the hud, and won't look good for all fonts. So the ideal is a matrix of strings that can be repositioned and respaced relatively, via separate draw calls.

FDAapproved committed:
Fix Menudef

Fixes a few parts of the menudef that weren't currently working (mostly text entry used)

FDAapproved commented:
Forgot to close this, so I'll do that really quick.

UndeadZeratul commented:
I remembered you still had that big data object refactor recently and I've been looking into reworking a bit of HHX soon, I'm thinking I may pull in that commit of yours onto a branch and try integrating it again, either into HHX or possibly as a suite of utility classes in HDCoreLib.  Was there anything else you wanted to include with them?
